src/main/java/ca/xqz/tweetmouth/Tweet.java

```diff
@@ -72,6 +101,13 @@ private void parseMessage() {
             references = tokens.parallelStream()
                     .filter(s -> s.startsWith("@") && s.length() > 1)
                     .map(s -> s.substring(1).toLowerCase())
+                    .map(s -> {
```

Comment:
I don't know exactly why there are trailing _s, but some of them have it even though they shouldn't. Having looked at the requirements of user handles, they cannot contain _. When we refine it a bit more, we can just remove non-conformant handles.